knife/lib/chef/knife/cookbook_upload.rb

```diff
@@ -163,6 +163,13 @@ def run
             end
           end
         end
+        { "status" => 200, "message" => "Success" } if @calling_request != "CLI"
```

Comment:
This isn't core to the knife cookbook upload command, so it seems very odd to have a CLI tool returning json  as if it were a single HTTP request handler.

Would it make more sense to have a command that wraps cookbook upload, and adds the necessary return information and error handling behavior?

Reply:
Yeah agree, CLI tool returning JSON looks odd to me also. Thanks for pointing it out.

I will make further changes accordingly.